### ISSUE TITLE
[BUG] Adding widgets crashes the app

### ISSUE THREAD

**Issue by benjaluth09** [Deletescape-Media/Lawnchair#3317]
### Describe the bug

When adding a widget from any app, the app crashes.
When coming back to the app, the widget appears, but when I try to resize it (by long-clicking it), it crashes.
I assume that's the bug, when trying to display the resize line.
Thanks for reading!

### Steps to reproduce

1. Long-click the homescreen
2. Click widget
3. Add a widget
4. Drag it to the homescreen
5. *Boom* The app crashes.

### Expected behavior

The app not crashing and displaying the resize line.

### Device information

- Device: [Amazon Fire Tablet HD 8]
- OS: [Android version 9.0, FireOS 7.3.2.6]
- App version: [12.1.0 Alpha 4]

### Additional context

Lawnchair bug report 8 feb. 2023 11:08:06
version: 12.1.0 Alpha 4 (12010005)
commit: b7ad08c
build.brand: Amazon
build.device: onyx
build.display: PS7326.3183N
build.fingerprint: Amazon/onyx/onyx:9/PS7326.3183N/0025602846464:user/amz-p,release-keys
build.hardware: mt8168
build.id: PS7326.3183N
build.manufacturer: Amazon
build.model: KFONWI
build.product: onyx
build.type: user
version.codename: REL
version.incremental: 0025602846596
version.release: 9
version.sdk_int: 28
display.density_dpi: 196
isRecentsEnabled: false

error: Uncaught exception

android.view.InflateException: Binary XML file line #28: Binary XML file line #28: Error inflating class <unknown>
Caused by: android.view.InflateException: Binary XML file line #28: Error inflating class <unknown>
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Constructor.newInstance0(Native Method)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
	at android.view.LayoutInflater.createView(LayoutInflater.java:647)
	at com.android.internal.policy.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:58)
	at android.view.LayoutInflater.onCreateView(LayoutInflater.java:720)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:788)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:863)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:866)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
	at com.android.launcher3.AppWidgetResizeFrame.showForWidget(AppWidgetResizeFrame.java:5)
	at com.android.launcher3.Launcher.completeAddAppWidget(Launcher.java:23)
	at com.android.launcher3.Launcher$4.run(Launcher.java:1)
	at com.android.launcher3.Workspace$6.run(Workspace.java:4)
	at com.android.launcher3.anim.AnimatorListeners$1.onAnimationEnd(Unknown Source:2)
	at android.animation.Animator$AnimatorListener.onAnimationEnd(Animator.java:552)
	at android.animation.AnimatorSet.endAnimation(AnimatorSet.java:1294)
	at android.animation.AnimatorSet.doAnimationFrame(AnimatorSet.java:1079)
	at android.animation.AnimationHandler.doAnimationFrame(AnimationHandler.java:146)
	at android.animation.AnimationHandler.access$100(AnimationHandler.java:37)
	at android.animation.AnimationHandler$1.doFrame(AnimationHandler.java:54)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:959)
	at android.view.Choreographer.doCallbacks(Choreographer.java:772)
	at android.view.Choreographer.doFrame(Choreographer.java:703)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:947)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:211)
	at android.app.ActivityThread.main(ActivityThread.java:6769)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
Caused by: android.content.res.Resources$NotFoundException: Drawable app.lawnchair:drawable/widget_resize_frame with resource ID #0x7f08031a
Caused by: android.content.res.Resources$NotFoundException: File res/pe.xml from drawable resource ID #0x7f08031a
	at android.content.res.ResourcesImpl.loadDrawableForCookie(ResourcesImpl.java:847)
	at android.content.res.ResourcesImpl.loadDrawable(ResourcesImpl.java:631)
	at android.content.res.Resources.loadDrawable(Resources.java:897)
	at android.content.res.TypedArray.getDrawableForDensity(TypedArray.java:955)
	at android.content.res.TypedArray.getDrawable(TypedArray.java:930)
	at android.widget.ImageView.<init>(ImageView.java:189)
	at android.widget.ImageView.<init>(ImageView.java:172)
	at android.widget.ImageView.<init>(ImageView.java:168)
	at java.lang.reflect.Constructor.newInstance0(Native Method)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
	at android.view.LayoutInflater.createView(LayoutInflater.java:647)
	at com.android.internal.policy.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:58)
	at android.view.LayoutInflater.onCreateView(LayoutInflater.java:720)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:788)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:863)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:866)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
	at com.android.launcher3.AppWidgetResizeFrame.showForWidget(AppWidgetResizeFrame.java:5)
	at com.android.launcher3.Launcher.completeAddAppWidget(Launcher.java:23)
	at com.android.launcher3.Launcher$4.run(Launcher.java:1)
	at com.android.launcher3.Workspace$6.run(Workspace.java:4)
	at com.android.launcher3.anim.AnimatorListeners$1.onAnimationEnd(Unknown Source:2)
	at android.animation.Animator$AnimatorListener.onAnimationEnd(Animator.java:552)
	at android.animation.AnimatorSet.endAnimation(AnimatorSet.java:1294)
	at android.animation.AnimatorSet.doAnimationFrame(AnimatorSet.java:1079)
	at android.animation.AnimationHandler.doAnimationFrame(AnimationHandler.java:146)
	at android.animation.AnimationHandler.access$100(AnimationHandler.java:37)
	at android.animation.AnimationHandler$1.doFrame(AnimationHandler.java:54)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:959)
	at android.view.Choreographer.doCallbacks(Choreographer.java:772)
	at android.view.Choreographer.doFrame(Choreographer.java:703)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:947)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:211)
	at android.app.ActivityThread.main(ActivityThread.java:6769)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
Caused by: java.lang.UnsupportedOperationException: Can't convert value at index 0 to dimension: type=0x3
	at android.content.res.TypedArray.getDimensionPixelSize(TypedArray.java:736)
	at android.graphics.drawable.GradientDrawable.updateDrawableCorners(GradientDrawable.java:1477)
	at android.graphics.drawable.GradientDrawable.inflateChildElements(GradientDrawable.java:1435)
	at android.graphics.drawable.GradientDrawable.inflate(GradientDrawable.java:1244)
	at android.graphics.drawable.DrawableInflater.inflateFromXmlForDensity(DrawableInflater.java:142)
	at android.graphics.drawable.Drawable.createFromXmlInnerForDensity(Drawable.java:1332)
	at android.graphics.drawable.Drawable.createFromXmlForDensity(Drawable.java:1291)
	at android.content.res.ResourcesImpl.loadDrawableForCookie(ResourcesImpl.java:833)
	... 42 more

**Comment by qeterme** [comment]
Same happens with Samsung widgets

```
Lawnchair bug report Feb 13, 2023 10:48:46 version: 12.1.0 Alpha 4 (12010005) commit: b7ad08c build.brand: samsung build.device: b0s build.display: TP1A.220624.014.S908BXXU3CWAI build.fingerprint: samsung/b0sxeea/b0s:13/TP1A.220624.014/S908BXXU3CWAI:user/release-keys build.hardware: s5e9925 build.id: TP1A.220624.014 build.manufacturer: samsung build.model: SM-S908B build.product: b0sxeea build.type: user version.codename: REL version.incremental: S908BXXU3CWAI version.release: 13 version.sdk_int: 33 display.density_dpi: 420 isRecentsEnabled: false error: Uncaught exception java.lang.NullPointerException: Attempt to read from field 'android.content.ComponentName android.appwidget.AppWidgetProviderInfo.provider' on a null object reference in method 'void com.android.launcher3.Launcher.completeAddAppWidget(int, com.android.launcher3.model.data.ItemInfo, android.appwidget.AppWidgetHostView, com.android.launcher3.widget.LauncherAppWidgetProviderInfo)' 	at com.android.launcher3.Launcher.completeAddAppWidget(Launcher.java:3) 	at com.android.launcher3.Launcher$4.run(Launcher.java:1) 	at com.android.launcher3.Workspace$6.run(Workspace.java:4) 	at com.android.launcher3.anim.AnimatorListeners$1.onAnimationEnd(Unknown Source:2) 	at android.animation.Animator$AnimatorListener.onAnimationEnd(Animator.java:600) 	at android.animation.AnimatorSet.endAnimation(AnimatorSet.java:1301) 	at android.animation.AnimatorSet.doAnimationFrame(AnimatorSet.java:1086) 	at android.animation.AnimationHandler.doAnimationFrame(AnimationHandler.java:307) 	at android.animation.AnimationHandler.-$$Nest$mdoAnimationFrame(Unknown Source:0) 	at android.animation.AnimationHandler$1.doFrame(AnimationHandler.java:86) 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1299) 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1309) 	at android.view.Choreographer.doCallbacks(Choreographer.java:923) 	at android.view.Choreographer.doFrame(Choreographer.java:847) 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1283) 	at android.os.Handler.handleCallback(Handler.java:942) 	at android.os.Handler.dispatchMessage(Handler.java:99) 	at android.os.Looper.loopOnce(Looper.java:226) 	at android.os.Looper.loop(Looper.java:313) 	at android.app.ActivityThread.main(ActivityThread.java:8757) 	at java.lang.reflect.Method.invoke(Native Method) 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571) 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067) 
```

**Comment by benjaluth09** [comment]
@qeterme It seems like it's failing to read the app package of the widget. (I assume)

**Comment by babyharpsea** [comment]
same issue here. posted it on discord

**Comment by benjaluth09** [comment]
@babyharpsea 😩 I'm dealing with this

**Comment by qeterme** [comment]
> @qeterme It seems like it's failing to read the app package of the widget. (I assume)

That's possible, I read on r/oneui that this issue happens with another launchers as well

**Comment by ajithrn** [comment]
dealing with same issue here on one ui 5. 1

**Comment by Scramzi** [comment]
same happened to me on one ui 2.5 android 10, and then it got fixed by itself? im using 12.10 dev #1415 btw

**Comment by IgnorantGirl** [comment]
same happened to me on start phone, so ｗhy is there an exception? What better solution?

03-29 06:52:00.254  9953  9953 E AndroidRuntime: java.lang.RuntimeException: Unable to create service com.android.quickstep.TouchInteractionService: android.view.InflateException: Binary XML file line #24 in com.android.launcher3:layout/taskbar: Binary XML file line #24 in com.android.launcher3:layout/taskbar: Error inflating class com.android.launcher3.taskbar.TaskbarView
03-29 06:52:00.254  9953  9953 E AndroidRuntime:         at android.app.ActivityThread.handleCreateService(ActivityThread.java:4725)
03-29 06:52:00.254  9953  9953 E AndroidRuntime:         at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
03-29 06:52:00.254  9953  9953 E AndroidRuntime:         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2314)
03-29 06:52:00.254  9953  9953 E AndroidRuntime:         at android.os.Handler.dispatchMessage(Handler.java:106)
03-29 06:52:00.254  9953  9953 E AndroidRuntime:         at android.os.Looper.loopOnce(Looper.java:216)
